pkg/kubelet/stats/pidlimit/pidlimit_linux.go

```diff
@@ -39,12 +41,33 @@ func Stats() (*statsapi.RlimitStats, error) {
 		}
 	}
 
-	var info syscall.Sysinfo_t
-	syscall.Sysinfo(&info)
-	procs := int64(info.Procs)
-	rlimit.NumOfRunningProcesses = &procs
+	if procs, err := runningProcNum(); err == nil {
+		rlimit.NumOfRunningProcesses = &procs
+	} else {
+		var info syscall.Sysinfo_t
+		syscall.Sysinfo(&info)
+		procs := int64(info.Procs)
+		rlimit.NumOfRunningProcesses = &procs
+	}
 
 	rlimit.Time = v1.NewTime(time.Now())
 
 	return rlimit, nil
 }
+
+func runningProcNum() (int64, error) {
```

Comment:
I would call this `countRunningTasks`. The number of running processes is typically much smaller than the number of tasks that this function actually returns (on Linux).
(yes, I know that the caller then stores this into a variable named `procs`)

Reply:
Another name could be `fetchCountOfRunningTasks`.